datatypes/src/main/java/org/hyperledger/besu/datatypes/Transaction.java

```diff
@@ -183,4 +187,126 @@ default Optional<? extends Quantity> getMaxFeePerBlobGas() {
    * @return the type of the transaction
    */
   TransactionType getType();
+
+  /**
+   * Boolean which indicates the transaction has associated cost data, whether gas price or 1559 fee
+   * market parameters.
+   *
+   * @return whether cost params are present
+   */
+  boolean hasCostParams();
+
+  /**
+   * Returns the number of blobs this transaction has, or 0 if not a blob transaction type
+   *
+   * @return return the count
+   */
+  int getBlobCount();
+
+  /**
+   * Returns the signature used to sign the transaction.
+   *
+   * @return the signature used to sign the transaction
+   */
+  SECPSignature getSignature();
```

Comment:
This can be derived from the `R`, `S`, and `V` fields.  Also, it presumes ECDSA forever.  If non-ECDSA keys are added we would want a more generic Signature type.